From a959cdf89b50887298cac0ae5b77a8588cfa3330 Mon Sep 17 00:00:00 2001
From: Beleswar Padhi <b-padhi@ti.com>
Date: Tue, 18 Mar 2025 15:49:54 +0530
Subject: [PATCH 3/5] remoteproc: k3-dsp: Prevent TI-SCI messaging

Currently, Device Manager (DM) is not up for tda5-vdk platform.
Therefore, prevent any TI-SCI messaging services in the K3 DSP
remoteproc driver. Hardcode the device state to "RPROC_DETACHED" to let
it boot in IPC-only mode currently.
---
 drivers/remoteproc/ti_k3_dsp_remoteproc.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/remoteproc/ti_k3_dsp_remoteproc.c b/drivers/remoteproc/ti_k3_dsp_remoteproc.c
index b6631b2ba6c0..a664a9e33586 100644
--- a/drivers/remoteproc/ti_k3_dsp_remoteproc.c
+++ b/drivers/remoteproc/ti_k3_dsp_remoteproc.c
@@ -81,9 +81,11 @@ static int k3_dsp_rproc_probe(struct platform_device *pdev)
 	if (!data)
 		return -ENODEV;
 
+	/*
 	ret = rproc_of_parse_firmware(dev, 0, &fw_name);
 	if (ret)
 		return dev_err_probe(dev, ret, "failed to parse firmware-name property\n");
+	*/
 
 	rproc = devm_rproc_alloc(dev, dev_name(dev), &k3_dsp_rproc_ops,
 				 fw_name, sizeof(*kproc));
@@ -105,6 +107,7 @@ static int k3_dsp_rproc_probe(struct platform_device *pdev)
 	if (ret)
 		return ret;
 
+	/*
 	kproc->ti_sci = devm_ti_sci_get_by_phandle(dev, "ti,sci");
 	if (IS_ERR(kproc->ti_sci))
 		return dev_err_probe(dev, PTR_ERR(kproc->ti_sci),
@@ -136,15 +139,21 @@ static int k3_dsp_rproc_probe(struct platform_device *pdev)
 	ret = k3_rproc_of_get_memories(pdev, kproc);
 	if (ret)
 		return ret;
+	*/
 
 	ret = k3_reserved_mem_init(kproc);
 	if (ret)
 		return dev_err_probe(dev, ret, "reserved memory init failed\n");
 
+	/*
 	ret = kproc->ti_sci->ops.dev_ops.is_on(kproc->ti_sci, kproc->ti_sci_id,
 					       NULL, &p_state);
 	if (ret)
 		return dev_err_probe(dev, ret, "failed to get initial state, mode cannot be determined\n");
+	*/
+
+	/* Hard-code to boot in IPC-only mode for now */
+	p_state = true;
 
 	/* configure devices for either remoteproc or IPC-only mode */
 	if (p_state) {
-- 
2.34.1

